app/services/irs_attempts_api/redis_client.rb

```diff
@@ -20,6 +20,8 @@ def write_event(event_key:, jwe:, timestamp:)
     def read_events(timestamp:)
       key = key(timestamp)
       redis_pool.with do |client|
+        # see client.hscan which refs https://redis.io/commands/scan/
+        # but it's... a lil' bit weird.
```

Comment:
Arguably, this change would be the big win, in allowing us to read and write in batches rather than fetching everything into memory. Will need to play around with this.

Reply:
I think a good task for tomorrow is to write something quick to bulk-generate events. We're going to want something like that for generating a large sample file for the IRS as well.

My gut feeling is that `hgetall` won't be a great choice with a huge number of events, and they'll all be loaded in memory. I like the idea of being able to `hscan` and write them to the file as we fetch them.

It is occurring to me tonight that we've talked about doing this "background processing" to fetch all the events, put them in a flat file, and then store that _in Redis_ rather than S3. Is that actually saving us anything over what we have now? All the more reason to generate a ton of events and bang on this.